compiler/rustc_ast_passes/src/ast_validation.rs

```diff
@@ -429,6 +439,21 @@ impl<'a> AstValidator<'a> {
         }
     }
 
+    fn check_foreign_item_safety(&self, item_span: Span, safety: Safety) {
+        let allow_unsafe_extern_blocks =
+            item_span.at_least_rust_2024() || self.features.unsafe_extern_blocks;
```

Comment:
From what I expected (and what TC wrote): we should only be checking the feature, not the edition.

Reply:
Per [my comment](https://github.com/rust-lang/rust/pull/124482#discussion_r1622986666), it would be OK, as far as edition management is concerned, to check for the feature **and** (not **or**) the edition.

It would also be OK to check only for the feature in this PR and to tie it to the edition (`&&`ed with the feature flag) as a second step.

Reply:
Actually, disregard.  Looking more closely on this diff, it should in fact just check for the feature flag.